In [2]:
import pandas as pd
import numpy as np
import scipy.stats

In [3]:
df = pd.read_csv("../Dataset/nyc-rolling-sales.csv")
df

Unnamed: 0  BOROUGH   NEIGHBORHOOD  \
0               4        1  ALPHABET CITY   
1               5        1  ALPHABET CITY   
2               6        1  ALPHABET CITY   
3               7        1  ALPHABET CITY   
4               8        1  ALPHABET CITY   
...           ...      ...            ...   
84543        8409        5        WOODROW   
84544        8410        5        WOODROW   
84545        8411        5        WOODROW   
84546        8412        5        WOODROW   
84547        8413        5        WOODROW   

                           BUILDING CLASS CATEGORY TAX CLASS AT PRESENT  \
0      07 RENTALS - WALKUP APARTMENTS                                2A   
1      07 RENTALS - WALKUP APARTMENTS                                 2   
2      07 RENTALS - WALKUP APARTMENTS                                 2   
3      07 RENTALS - WALKUP APARTMENTS                                2B   
4      07 RENTALS - WALKUP APARTMENTS                                2A   
...                                            ...                  ...   
84543  02 TWO FAMILY DWELLINGS                                        1   
84544  02 TWO FAMILY DWELLINGS                                        1   
84545  02 TWO FAMILY DWELLINGS                                        1   
84546  22 STORE BUILDINGS                                             4   
84547  35 INDOOR PUBLIC AND CULTURAL FACILITIES                       4   

       BLOCK  LOT EASE-MENT BUILDING CLASS AT PRESENT                 ADDRESS  \
0        392    6                                  C2            153 AVENUE B   
1        399   26                                  C7   234 EAST 4TH   STREET   
2        399   39                                  C7   197 EAST 3RD   STREET   
3        402   21                                  C4     154 EAST 7TH STREET   
4        404   55                                  C2  301 EAST 10TH   STREET   
...      ...  ...       ...                       ...                     ...   
84543   7349   34                                  B9           37 QUAIL LANE   
84544   7349   78                                  B9        32 PHEASANT LANE   
84545   7351   60                                  B2        49 PITNEY AVENUE   
84546   7100   28                                  K6   2730 ARTHUR KILL ROAD   
84547   7105  679                                  P9       155 CLAY PIT ROAD   

       ... RESIDENTIAL UNITS  COMMERCIAL UNITS  TOTAL UNITS  LAND SQUARE FEET  \
0      ...                 5                 0            5              1633   
1      ...                28                 3           31              4616   
2      ...                16                 1           17              2212   
3      ...                10                 0           10              2272   
4      ...                 6                 0            6              2369   
...    ...               ...               ...          ...               ...   
84543  ...                 2                 0            2              2400   
84544  ...                 2                 0            2              2498   
84545  ...                 2                 0            2              4000   
84546  ...                 0                 7            7            208033   
84547  ...                 0                 1            1             10796   

       GROSS SQUARE FEET YEAR BUILT TAX CLASS AT TIME OF SALE  \
0                   6440       1900                         2   
1                  18690       1900                         2   
2                   7803       1900                         2   
3                   6794       1913                         2   
4                   4615       1900                         2   
...                  ...        ...                       ...   
84543               2575       1998                         1   
84544               2377       1998                         1   
84545               1496       192

========================================================================================================================

# Measure of Central Tendency

----------------------------------------------------------------------------------------------------------------------------

### Mean

Pertanyaan : **Harga rata-rata Apartemen di kota New York**

Kolom yang dipakai = 'SALE PRICE'
- Sebelum dilakukan pencarian nilai rata-rata pada kolom sale price, nilai missing pada kolom tersebut harus diatasi terlebih dahulu

In [4]:
# Kode baris pertama berguna untuk mengubah data '-' pada kolom SALE PRICE menjadi 0
# Nilai 0 digunakan untuk mengganti nilai '-' dikarenakan nilai tersebut tidak akan mengganggu nilai MEAN data yang telah tertulis

mean_list = df['SALE PRICE']
mean_list = mean_list.replace([' -  '], '0')
mean_list = mean_list.astype(float)


In [5]:
mean = mean_list.mean()
print('Rata - rata harga bangunan di kota New York = $', round(mean))

Rata - rata harga bangunan di kota New York = $ 1056623


Di kota New York harga rata-rata Apartemen yaitu sebesar **1.056.623 dolar**, dengan ketentuan : hanya menghitung bangunan yang memiliki harga bangunan pada dataset

----------------------------------------------------------------------------------------------------------------------------

### Median

Pertanyaan : **Median dari beberepa kolom Numerikal pada dataframe**

In [6]:
median_series = [df['BLOCK'], df['LOT'], df['RESIDENTIAL UNITS'], df['COMMERCIAL UNITS'], df['TOTAL UNITS']]
name = ['BLOCK', 'LOT', 'RESIDENTIAL UNITS', 'COMMERCIAL UNITS', 'TOTAL UNITS']

In [7]:
for i in range(len(median_series)):
    x = np.median(median_series[i])    
    print('Median dari kolom ', name[i], ' = ', x)


Median dari kolom  BLOCK  =  3311.0
Median dari kolom  LOT  =  50.0
Median dari kolom  RESIDENTIAL UNITS  =  1.0
Median dari kolom  COMMERCIAL UNITS  =  0.0
Median dari kolom  TOTAL UNITS  =  1.0


----------------------------------------------------------------------------------------------------------------------------

### Modus
Modus dari beberepa kolom kategorikal pada dataframe

Pertanyaan : 
- Neighborhood yang memiliki bangunan terbanyak
- Bangunan dengan kelas pajak terbanyak
- Borough/wilayah dengan ketersediaan bangunan terbanyak

In [8]:
mode_series = [df['NEIGHBORHOOD'], df['TAX CLASS AT PRESENT'], df['BOROUGH']]

In [9]:
# Pertanyaan 1 : Neighborhood yang memiliki bangunan terbanyak

mode1 = scipy.stats.mode(mode_series[0])
print('Jawaban pertanyaan 1 : ', mode1.mode , mode1.count)

Jawaban pertanyaan 1 :  ['FLUSHING-NORTH'] [3068]


Neighborhood di kota New York yang memiliki apartemen terbanyak yaitu di Neighborhood **FLUSHING NORTH** yang memiliki jumlah apartemen sebanyak **3068 Bangunan**

In [10]:
# Pertanyaan 2 : Bangunan dengan kelas pajak terbanyak

mode2 = scipy.stats.mode(mode_series[1])
print('Jawaban pertanyaan 2 : ', mode2.mode , mode2.count)

Jawaban pertanyaan 2 :  ['1'] [38633]


Di kota New York banyak bangunan yang memiliki pajak bangunan **kelas 1** dengan jumlah bangunan sebanyak **38633 Bangunan**, 

Artinya kebanyakan apartemen di kota New York memiliki pajak yang cukup terjangkau bagi masyarakat kelas menengah keatas

In [11]:
# Pertanyaan 3 : Borough/wilayah dengan ketersediaan bangunan terbanyak

mode3 = scipy.stats.mode(mode_series[2])
print('Jawaban pertanyaan 3 : ', mode3.mode , mode3.count)

Jawaban pertanyaan 3 :  [4] [26736]


Wilayah **Queens (4)** merupakan wilayah terpadat dikarenakan memiliki jumlah apartemen terbanyak dibandingkan dengan wilayah lain di kota New York yaitu sebanyak **26736 Bangunan**

========================================================================================================================

# Measure of Spread

### Range

Pertanyaan : **Range dari Harga Apartemen termahal dengan Harga apartemen termurah**

In [12]:
df['SALE PRICE'] = df['SALE PRICE'].replace([' -  '], '0')
df['SALE PRICE'] = df['SALE PRICE'].astype(float)

In [18]:
ranged = df
ranged.drop(ranged[ranged['SALE PRICE'] <= 0].index, inplace=True)

In [20]:
print("Harga Apartemen Terendah  : $ ", np.amin(ranged['SALE PRICE']))
print("Harga Apartemen Tertinggi : $ ", np.amax(ranged['SALE PRICE']))
print("Jarak antar Harga Apartemen Terendah dan tertinggi : $ ", np.ptp(ranged['SALE PRICE']))

Harga Apartemen Terendah  : $  1.0
Harga Apartemen Tertinggi : $  2210000000.0
Jarak antar Harga Apartemen Terendah dan tertinggi : $  2209999999.0


Jawab : Jarak antara harga apartemen tertinggi dengan terendah adalah $ **2209999999.0**

----------------------------------------------------------------------------------------------------------------------------

### Variance

Pertanyaan : **Variance dari kolom harga apartemen**

In [21]:
var = np.var(ranged['SALE PRICE'])
print('Variance : ', var)

Variance :  152014779013418.53


Jawab : Variance nya adalah **152014779013418.53**

----------------------------------------------------------------------------------------------------------------------------

### Standard Deviation

Pertanyaan : **Standard Deviation dari kolom harga apartemen**

In [24]:
std = var ** 0.5
print('Standard Deviation : ', round(std))

Standard Deviation :  12329427


Jawab : Standard Deviation dari Kolom harga apartemen adalah akar kuadrat dari nilai variance nya, nilainya : **12329427**

========================================================================================================================

# Probability Distribution